### PR TITLE
MEN-4779 introduce enterprise version of the deviceauth service

### PR DESCRIPTION
**Pull request by MaciejTe**
Introduced enterprise version of deviceauth service in release tool and backend tests, updated  enterprise docker compose file

Changelog: None

Signed-off-by: Maciej Tomczuk <maciej.tomczuk@northern.tech>

### CODE REVIEW

**Review by kacf — APPROVED** [review]

**Comment by MaciejTe** [comment]
@mender-test-bot  start pipeline

File: testutils/infra/cli.py [hunk 1]
```diff
@@ -12,12 +12,16 @@
...
 
+from collections import namedtuple
+
 from testutils.infra.container_manager.docker_manager import DockerNamespace
 from testutils.infra.container_manager.kubernetes_manager import (
     KubernetesNamespace,
     isK8S,
 )
 
+Service = namedtuple("Service", "bin_path data_path")
```
Comment (lluiscampos) [inline testutils/infra/cli.py:23]: Nitpick: I would name this `Microservice` or `Serverservice`, to clearly differentiate from "systemd service" concept that we have in this test suite

File: testutils/infra/cli.py [hunk 2]
```diff
@@ -121,15 +125,28 @@ def migrate(self):
...
+        enterprise = Service(
+            "/usr/bin/deviceauth-enterprise", "/etc/deviceauth-enterprise"
+        )
+
+        for service in [open_source, enterprise]:
+            try:
+                self.container_manager.execute(self.cid, [service.bin_path, "--help"])
+                self.service = service
+            except:
+                continue
```
Comment (lluiscampos) [inline testutils/infra/cli.py:143]: Looks good, but to make we have found at least one valid binary:
```suggestion
        for service in [open_source, enterprise]:
            try:
                self.container_manager.execute(self.cid, [service.bin_path, "--help"])
                self.service = service
                break
            except:
                continue
        else:
            raise RuntimeError("no runnable binary found in mender-deviceauth")
```

Comment (lluiscampos) [reply testutils/infra/cli.py:143]: Alternatively, as done in other services, something like:
```
        if self.service is None:
            raise RuntimeError("no runnable binary found in mender-useradm")
```

But I personally prefer the former.

File: testutils/infra/cli.py [hunk 3]
```diff
@@ -144,12 +161,12 @@ def add_default_tenant_token(self, tenant_token):
         :param tenant_token - 'the default tenant token to set'
         """
 
-        # Append the default_tenant_token in the config ('/etc/deviceauth/config.yaml')
+        # Append the default_tenant_token in the config (f'{self.service.data_path}/config.yaml')
```
Comment (lluiscampos) [inline testutils/infra/cli.py:164]: Maybe the comment does not need the path explicitly, it is four lines below anyway...

**Review by lluiscampos — APPROVED** [review]
Nice addition with the `namedtuple` :+1:

File: testutils/infra/cli.py (hunk 2 above)
Comment (MaciejTe) [reply testutils/infra/cli.py:143]: Added RuntimeError, thanks!

**Review by lluiscampos — CHANGES_REQUESTED** [review]
See comment above

File: git-versions-enterprise.yml [hunk 4]
```diff
@@ -33,5 +33,6 @@ services:
 
     mender-devicemonitor:
         image: registry.mender.io/mendersoftware/devicemonitor:master
-
-
+
```
[the remark this answers is not in the document]
Comment (MaciejTe) [reply git-versions-enterprise.yml]: done, thanks

**Pull request merged**